app-editors/atom/atom-1.27.2.ebuild

```diff
@@ -0,0 +1,563 @@
+
+	gypopts="--verbose"
+
+	if [[ ${MAKEOPTS} == *-j* && ${jobs} != 999 ]]; then
+		gypopts+=" --jobs ${jobs}"
+	fi
+
+	mkdir -p "${BUILD_DIR}/modules/" || die
+
+	for binmod in ${BINMODS[@]}; do
+		einfo "Building ${binmod}..."
+		cd "${WORKDIR}/$(package_dir ${binmod})" || die
+		enodegyp_atom ${gypopts} build
+		x=${binmod##node-}
+		mkdir -p "${BUILD_DIR}/modules/${x}" || die
+		cp build/Release/*.node "${BUILD_DIR}/modules/${x}" || die
+	done
+
+	# Put compiled binary modules in place
+	_fix_binmods "${BUILD_DIR}/app" "apm"
+	_fix_binmods "${BUILD_DIR}/app" "node_modules"
+
+	# Remove non-Linux vendored ctags binaries
+	rm "${BUILD_DIR}/app/${ctags_d}/ctags-darwin" \
+	   "${BUILD_DIR}/app/${ctags_d}/ctags-win32.exe" || die
+
+	# Re-pack app.asar
+	# Keep unpack rules in sync with build/tasks/generate-asar-task.coffee
+	cd "${BUILD_DIR}" || die
+	x="--unpack={*.node,ctags-config,ctags-linux,**/spec/fixtures/**,**/node_modules/spellchecker/**,**/resources/atom.png}"
+	xd="--unpack-dir=apm"
+	easar pack "${x}" "${xd}" "app" "app.asar"
+
+	rm -r "${BUILD_DIR}/app.asar.unpacked/apm" || die
+
+	# Replace vendored ctags with a symlink to system ctags
+	rm "${BUILD_DIR}/app.asar.unpacked/${ctags_d}/ctags-linux" || die
+	ln -s "${EROOT%/}/usr/bin/ctags" \
+		"${BUILD_DIR}/app.asar.unpacked/${ctags_d}/ctags-linux" || die
+}
+
+src_test() {
+	local electron="$(get_electron_dir)/electron"
+	local app="${BUILD_DIR}/app.asar"
+
+	virtx "${electron}" --app="${app}" --test "${app}/spec"
+}
+
+src_install() {
+	local install_dir="$(get_install_dir)"
+	local suffix="$(get_install_suffix)"
+
+	insinto "${install_dir}"
+
+	doins "${BUILD_DIR}/app.asar"
+	doins -r "${BUILD_DIR}/app.asar.unpacked"
+
+	insinto "${install_dir}/app"
+	doins -r "${BUILD_DIR}/app/apm"
+
+	insinto "/usr/share/applications/"
+	newins "${BIN_S}/usr/share/applications/$(get_atom_appname).desktop" \
+		"atom${suffix}.desktop"
+
+	insinto "/usr/share/icons/"
+	doins -r "${BIN_S}/usr/share/icons/hicolor"
+
+	exeinto "${install_dir}"
+	newexe "${BUILD_DIR}/app/atom.sh" atom
+	insinto "/usr/share/licenses/${PN}${suffix}"
+	doins "${BIN_S}/$(get_atom_rpmdir)/resources/LICENSE.md"
+	dosym "${install_dir}/atom" "/usr/bin/atom${suffix}"
+	dosym "${install_dir}/app/apm/bin/apm" "/usr/bin/apm${suffix}"
+
+	_fix_executables "${install_dir}/app/apm/bin"
+	_fix_executables "${install_dir}/app/apm/node_modules/.bin"
+	_fix_executables "${install_dir}/app/apm/node_modules/npm/bin"
+	_fix_executables "${install_dir}/app/apm/node_modules/npm/bin/node-gyp-bin"
+	_fix_executables "${install_dir}/app/apm/node_modules/node-gyp/bin"
+}
+
+pkg_postinst() {
+	xdg_desktop_database_update
+}
+
+pkg_postrm() {
+	xdg_desktop_database_update
+}
+
+# Helpers
+# -------
+
+# Return the installation suffix appropriate for the slot.
+get_install_suffix() {
+	local c=(${SLOT//\// })
+	local slot=${c[0]}
+	local suffix
+
+	if [[ "${slot}" == "0" ]]; then
+		suffix=""
+	else
+		suffix="-${slot}"
+	fi
+
+	echo -n "${suffix}"
```

Comment:
No need for `-n`, bash implicitly strips newline when using command substitution. Applies to all below.